gp-settings.php

```diff
@@ -150,15 +184,21 @@ function gp_rewrite_rules() {
 		// When GlotPress is set to take over the root of the site,
 		// add a special rule that WordPress uses to route requests to root.
 		add_rewrite_rule( '$', 'index.php?gp_route', 'top' );
-
-		$match_regex = '^(.*)$';
-	} else {
-		$match_regex = '^' . $gp_base . '/?(.*)$';
 	}
 
+	$match_regex = gp_generate_rewrite_rules();
+	
 	add_rewrite_rule( $match_regex, 'index.php?gp_route=$matches[1]', 'top' );
+
+	// Check to see if the rewrite rule has changed, if so, update the option and flush the rewrite rules.
+	if( $match_regex != get_option( 'gp_rewrite_rule' ) ) {
+		// Save the rewrite rule to an option so we have something to compare against later.  We don't need to worry about the root rewrite rule above as it is always the same.
+		update_option( 'gp_rewrite_rule', $match_regex );
+		flush_rewrite_rules();
+	}
 }
-add_action( 'init', 'gp_rewrite_rules' );
+// Since we may be flushing the rewrite rules move our priority to to the end to make sure other plugins have a chance to setup thier own rules.
+add_action( 'init', 'gp_rewrite_rules', 1000 );
```

Comment:
This shouldn't be necessary, see https://core.trac.wordpress.org/changeset/31104.